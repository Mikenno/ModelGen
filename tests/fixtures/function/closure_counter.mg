
func counter(n = 0, delta = 1)
	return func(d = delta)
		n += d
		return n

c1 = counter()
c2 = counter(10)
c3 = counter(10, 5)

print(c1())
print(c1())
print(c1(10))
print()

print(c2())
print(c2())
print(c2(100))
print()

print(c3())
print(c3())
print(c3(1))
print()

print(c1())
print(c1())
print()

print(c2())
print(c2())
print()

print(c3())
print(c3())
print()
