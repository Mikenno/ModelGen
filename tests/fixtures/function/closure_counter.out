1
2
12

11
12
112

15
20
21

13
14

113
114

26
31

